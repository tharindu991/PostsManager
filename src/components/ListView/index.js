/*
 * Author: Tharindu Seneviratne
 * Date: 29/02/2020
 *
 * List view with component
 */
import React from 'react';
import {View, FlatList} from 'react-native';
import PropTypes from 'prop-types';
import {StyleSheet} from 'react-native';

const ListView = ({renderItem, data = [], onRefresh, refreshing}) => {
  return (
    <View style={styles.container}>
      <FlatList
        data={data}
        renderItem={({item}) => renderItem(item)}
        keyExtractor={item => `${item.id}`}
        onRefresh={onRefresh}
        refreshing={refreshing}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});

ListView.propTypes = {
  data: PropTypes.array.isRequired,
  renderItem: PropTypes.func.isRequired,
  refreshing: PropTypes.bool.isRequired,
  onRefresh: PropTypes.func.isRequired,
};

export default ListView;
