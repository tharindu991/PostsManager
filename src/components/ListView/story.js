/*
 * Author: Tharindu Seneviratne
 * Date: 29/02/2020
 *
 * Defines stories for ListView component
 */

import React from 'react';
import {storiesOf} from '@storybook/react-native';
import ListView from './';
import {Text, View} from 'react-native';

const Container = ({children}) => (
  <View
    style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: '#dce3e5',
    }}>
    {children}
  </View>
);

storiesOf('ListView', module).add('default view', () => (
  <Container>
    <ListView
      data={[
        {
          id: 5,
          title: 'title',
          body: 'body',
          user: 'user',
        },
      ]}
      renderItem={item => <Text>{item.user}</Text>}
      refreshing={false}
      onRefresh={() => console.log('onRefresh: ListView onRefresh called')}
    />
  </Container>
));
