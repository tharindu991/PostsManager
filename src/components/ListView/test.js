/*
 * Author: Tharindu Seneviratne
 * Date: 29/02/2020
 *
 * Defines tests for ListView component
 */
import React from 'react';
import ListView from './';
import renderer from 'react-test-renderer';
import {Text} from 'react-native';
const onRefresh = jest.fn();

describe('ListView Component', () => {
  it('ListView component renders correctly', () => {
    const component = renderer
      .create(
        <ListView
          data={[
            {
              id: 1,
              title: 'title',
              body: 'body',
              user: 'user',
            },
          ]}
          renderItem={item => <Text>{item.user}</Text>}
          refreshing={false}
          onRefresh={onRefresh}
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
