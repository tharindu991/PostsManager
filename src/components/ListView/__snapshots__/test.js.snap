// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView Component ListView component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "body": "body",
          "id": 1,
          "title": "title",
          "user": "user",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[MockFunction]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[MockFunction]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text>
          user
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
